pyuvdata/uvbeam/uvbeam.py

```diff
@@ -4018,11 +4022,12 @@ def read_cst_beam(
                 "model_name": model_name,
                 "model_version": model_version,
                 "history": history,
+                "thetaphi_to_azza_fnc": thetaphi_to_azza_fnc,
             }
             if "ref_imp" in settings_dict:
                 overriding_keywords["reference_impedance"] = reference_impedance
             if "x_orientation" in settings_dict:
-                overriding_keywords["x_orientation"] = reference_impedance
+                overriding_keywords["x_orientation"] = x_orientation
```

Comment:
oof, thanks for finding and fixing this!

Reply:
👍🏻